### ISSUE TITLE
Commander hopefully no longer flaky

### ISSUE THREAD

**Issue by shadowspawn** [nodejs/citgm#838]
There was a timing issue with Commander's tests that caused it to be marked as flaky in `citgm` in Dec 2019: https://github.com/nodejs/citgm/pull/773

An issue got opened against Commander, and improvements were made in Commander v4.1 in Jan 2020: https://github.com/tj/commander.js/issues/1123

The commander entry in `citgm` was not updated, and was subsequently disabled in Mar 2020: https://github.com/nodejs/citgm/pull/790

Hopefully Commander is good to add back into the tests?

(I didn't find any evidence Commander was retried. Happy to help, but unfamiliar with `citgm` details.)

**Comment by shadowspawn** [comment]
```shell
citgm % bin/citgm.js commander                                                                                                                           main
info:    starting            | commander           
info:    lookup              | commander           
info:    lookup-found        | commander           
info:    commander lookup-replace| https://github.com/tj/commander.js/archive/0192d6664a0d52af54ba73e83b8527faa38508e8.tar.gz
info:    commander npm:      | Downloading project: https://github.com/tj/commander.js/archive/0192d6664a0d52af54ba73e83b8527faa38508e8.tar.gz
info:    commander npm:      | Project downloaded commander-6.2.0.tgz
info:    commander npm:      | npm install started 
info:    commander npm:      | npm install successfully completed
info:    commander npm:      | test suite started  
info:    passing module(s)   |                     
info:    module name:        | commander           
info:    version:            | 6.2.0               
info:    done                | The smoke test has passed.
info:    duration            | test duration: 25924ms
```

**Comment by richardlau** [comment]
CI of `citgm commander`: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/130/

Looks like it passes everywhere except on Windows, e.g.
<details>

<summary>With Node.js 10:</summary>

https://ci.nodejs.org/job/citgm-smoker-nobuild/987/nodes=win2016-vs2017/testReport/junit/(root)/citgm/commander_v6_2_0/

```
 undefinedadded 748 packages from 417 contributors in 30.622s
 > commander@6.2.0 test C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander
 > jest && npm run test-typings
 ----------|---------|----------|---------|---------|--------------------------------------------------------------------------------------------------------------------------------
 File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                                                                                              
 ----------|---------|----------|---------|---------|--------------------------------------------------------------------------------------------------------------------------------
 All files |   91.08 |    86.84 |   96.15 |   90.78 |                                                                                                                                
  index.js |   91.08 |    86.84 |   96.15 |   90.78 | 391,685,756,765,791,794,850,858,864,873-875,883-890,903-904,913,921-935,1014,1213,1257,1328,1481,1524,1588,1636,1753,1850-1879 
 ----------|---------|----------|---------|---------|--------------------------------------------------------------------------------------------------------------------------------
 PASS tests/options.mandatory.test.js (6.166 s)
 PASS tests/command.exitOverride.test.js (6.244 s)
 PASS tests/command.help.test.js
 PASS tests/options.variadic.test.js
 PASS tests/command.parseOptions.test.js (6.791 s)
 PASS tests/options.custom-processing.test.js
 PASS tests/options.version.test.js
 PASS tests/command.action.test.js
 PASS tests/options.bool.combo.test.js
 PASS tests/options.bool.test.js
 PASS tests/helpwrap.test.js
 PASS tests/command.asterisk.test.js
 PASS tests/command.helpOption.test.js
 PASS tests/command.helpCommand.test.js
 PASS tests/options.values.test.js
 FAIL tests/command.default.test.js
    default executable command  when default subcommand and no command then call default
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    default executable command  when default subcommand and unrecognised argument then call default with argument
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm an-argument
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    default executable command  when default subcommand and unrecognised option then call default with option
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm --an-option
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
 PASS tests/options.required.test.js
 FAIL tests/command.executableSubcommand.lookup.test.js
    when subcommand file has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm install
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when alias subcommand file has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm i
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand target executablefile has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm specifyInstall
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand file suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm publish
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when alias subcommand file suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm p
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand target executablefile has suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm specifyPublish
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand suffix is .ts then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures-ts\pm.ts install
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subsubcommand then lookup sub-sub-command
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm cache clear
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
 PASS tests/options.opts.test.js
 PASS tests/options.flags.test.js
 PASS tests/command.usage.test.js
 PASS tests/commander.configureCommand.test.js
 PASS tests/args.variadic.test.js
 PASS tests/command.alias.test.js
 PASS tests/command.parse.test.js
 PASS tests/options.optional.test.js
 PASS tests/command.unknownOption.test.js
 PASS tests/command.addCommand.test.js
 PASS tests/command.allowUnknownOptions.test.js
 PASS tests/options.camelcase.test.js
 PASS tests/command.unknownCommand.test.js
 PASS tests/options.regex.test.js
 PASS tests/args.literal.test.js
 PASS tests/options.detectClash.test.js
 PASS tests/createCommand.test.js
 PASS tests/command.name.test.js
 PASS tests/command.commandHelp.test.js
 PASS tests/command.executableSubcommand.test.js
 PASS tests/program.test.js
 PASS tests/options.bool.small.combined.test.js
 PASS tests/command.nested.test.js
 PASS tests/command.description.test.js
 PASS tests/command.executableSubcommand.inspect.test.js
 Summary of all failing tests
 FAIL tests/command.default.test.js
    default executable command  when default subcommand and no command then call default
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    default executable command  when default subcommand and unrecognised argument then call default with argument
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm an-argument
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    default executable command  when default subcommand and unrecognised option then call default with option
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm --an-option
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
 FAIL tests/command.executableSubcommand.lookup.test.js
    when subcommand file has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm install
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when alias subcommand file has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm i
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand target executablefile has no suffix then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm specifyInstall
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand file suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm publish
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when alias subcommand file suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm p
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand target executablefile has suffix .js then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm specifyPublish
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subcommand suffix is .ts then lookup succeeds
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures-ts\pm.ts install
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
    when subsubcommand then lookup sub-sub-command
     Command failed: node C:\Program Files\Git\citgm_tmp\e830a03c-81df-4757-846a-ab8f041f799a\commander\tests\fixtures\pm cache clear
     internal/modules/cjs/loader.js:638
         throw err;
         ^
     Error: Cannot find module 'C:\Program'
       at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
       at Function.Module._load (internal/modules/cjs/loader.js:562:25)
       at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
       at startup (internal/bootstrap/node.js:283:19)
       at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
 Test Suites: 2 failed, 1 skipped, 41 passed, 43 of 44 total
 Tests:       11 failed, 7 skipped, 353 passed, 371 total
 Snapshots:   0 total
 Time:        20.971 s
 Ran all test suites.
 npm ERR! Test failed.  See above for more details.
```
</details>

Looks like the tests fail if run in a directory containing spaces?

I'd also approve a PR that skipped Windows for now and enabled the tests everywhere else.
The skip info is in [`lookup.json`](https://github.com/nodejs/citgm/blob/main/lib/lookup.json), e.g. for `commander`:
https://github.com/nodejs/citgm/blob/b301249e95e86efbd18c4dfb54e3237ee85317ec/lib/lookup.json#L99-L103

**Comment by shadowspawn** [comment]
Thanks, I'll look into the failure and then do a PR.

**Comment by shadowspawn** [comment]
Fix in Commander has been released in v6.2.1 so unblocked.
